samples/bluetooth/hap_ha/src/vcp_vol_renderer.c

```diff
@@ -0,0 +1,180 @@
+		printk("AICS input type get failed (%d) for inst %p\n", err, inst);
+	} else {
+		printk("AICS inst %p input type %u\n", inst, input_type);
+	}
+}
+
+static void aics_status_cb(struct bt_aics *inst, int err, bool active)
+{
+	if (err) {
+		printk("AICS status get failed (%d) for inst %p\n", err, inst);
+	} else {
+		printk("AICS inst %p status %s\n", inst, active ? "active" : "inactive");
+	}
+
+}
+static void aics_description_cb(struct bt_aics *inst, int err, char *description)
+{
+	if (err) {
+		printk("AICS description get failed (%d) for inst %p\n", err, inst);
+	} else {
+		printk("AICS inst %p description %s\n", inst, description);
+	}
+}
+static void vocs_state_cb(struct bt_vocs *inst, int err, int16_t offset)
+{
+	if (err) {
+		printk("VOCS state get failed (%d) for inst %p\n", err, inst);
+	} else {
+		printk("VOCS inst %p offset %d\n", inst, offset);
+	}
+}
+
+static void vocs_location_cb(struct bt_vocs *inst, int err, uint32_t location)
+{
+	if (err) {
+		printk("VOCS location get failed (%d) for inst %p\n", err, inst);
+	} else {
+		printk("VOCS inst %p location %u\n", inst, location);
+	}
+}
+
+static void vocs_description_cb(struct bt_vocs *inst, int err, char *description)
+{
+	if (err) {
+		printk("VOCS description get failed (%d) for inst %p\n", err, inst);
+	} else {
+		printk("VOCS inst %p description %s\n", inst, description);
+	}
+}
+
+static struct bt_vcs_cb vcs_cbs = {
+	.state = vcs_state_cb,
+	.flags = vcs_flags_cb,
+};
+
+static struct bt_aics_cb aics_cbs = {
+	.state = aics_state_cb,
+	.gain_setting = aics_gain_setting_cb,
+	.type = aics_input_type_cb,
+	.status = aics_status_cb,
+	.description = aics_description_cb
+};
+
+static struct bt_vocs_cb vocs_cbs = {
+	.state = vocs_state_cb,
+	.location = vocs_location_cb,
+	.description = vocs_description_cb
+};
+
+int vcp_vol_renderer_init(void)
+{
+	int err;
+	struct bt_vcs_register_param vcs_param;
+	char input_desc[CONFIG_BT_VCS_AICS_INSTANCE_COUNT][16];
+	char output_desc[CONFIG_BT_VCS_VOCS_INSTANCE_COUNT][16];
+
+	memset(&vcs_param, 0, sizeof(vcs_param));
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_param.vocs_param); i++) {
+		vcs_param.vocs_param[i].location_writable = true;
+		vcs_param.vocs_param[i].desc_writable = true;
+		snprintf(output_desc[i], sizeof(output_desc[i]), "Output %d", i + 1);
+		vcs_param.vocs_param[i].output_desc = output_desc[i];
+		vcs_param.vocs_param[i].cb = &vocs_cbs;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_param.aics_param); i++) {
+		vcs_param.aics_param[i].desc_writable = true;
+		snprintf(input_desc[i], sizeof(input_desc[i]), "Input %d", i + 1);
+		vcs_param.aics_param[i].description = input_desc[i];
+		vcs_param.aics_param[i].type = BT_AICS_INPUT_TYPE_UNSPECIFIED;
+		vcs_param.aics_param[i].status = true;
+		vcs_param.aics_param[i].gain_mode = BT_AICS_MODE_MANUAL;
+		vcs_param.aics_param[i].units = 1;
+		vcs_param.aics_param[i].min_gain = -100;
+		vcs_param.aics_param[i].max_gain = 100;
+		vcs_param.aics_param[i].cb = &aics_cbs;
+	}
```

Comment:
Is AICS used for HA?

Reply:
Yes, here's quote from HAP spec:
"An HA may instantiate one or more instances of Audio Input Control Service (AICS) to expose control of
the gain of its inputs to a Volume Controller."